### PR TITLE
CI: fix FreeBSD CI passed when tests failed

### LINKED ISSUE #3779: FreeBSD: the CI should fail when the build fails or the tests are failing
In the CI, for example here:
https://github.com/uutils/coreutils/runs/7686143875?check_suite_focus=true

the tests are failing:
https://github.com/uutils/coreutils/issues/3778

And the build was failing:
https://github.com/uutils/coreutils/issues/3733

but they were both green in the CI

### PR DESCRIPTION
**Pull request by miles170**
Closes #3779.

Check that FreeBSD CI failed when the builds or tests failed here: https://github.com/miles170/coreutils/actions/runs/3488640374/jobs/5837818835

### CODE REVIEW

**Comment by sylvestre** [comment]
> test_cp: fix failing tests for FreeBSD

AFAIK, you are ignoring the tests, not fixing them :)
Could you please update the commit message ?

besides that, it looks good

**Comment by miles170** [comment]
The following tests still failed on FreeBSD. Should it be fixed or ignored on this PR?

```
  failures:
      test_cp::test_copy_contents_fifo
      test_ls::test_ls_io_errors
```

**Comment by miles170** [comment]
The following tests still failed on FreeBSD. Should it be fixed or ignored on this PR?

```
  failures:
      test_cp::test_copy_contents_fifo
      test_ls::test_ls_io_errors
```

**Comment by miles170** [comment]
`coreutils cp` calls `std::fs::cp` underling on FreeBSD, and `std::fs::cp` doesn't support copy `fifo`.

```sh
test@freebsd:~ $ mkfifo test
test@freebsd:~ $ /usr/home/miles/coreutils/target/debug/coreutils cp test outfile
cp: 'test' -> 'outfile': the source path is neither a regular file nor a symlink to a regular file
```

**Comment by miles170** [comment]
```sh
miles@freebsd:~/coreutils % cargo test --features unix -p uucore
error: none of the selected packages contains these features: unix, did you mean: utf8, utmpx?
```

File: tests/by-util/test_cp.rs [hunk 1]
```diff
@@ -2310,9 +2310,15 @@ fn test_same_file_force_backup() {
...
-#[cfg(unix)]
+#[cfg(all(unix, not(target_os = "freebsd")))]
 #[test]
 fn test_copy_contents_fifo() {
+    // TODO this test should work on FreeBSD, but the command was
+    // causing an error:
+    //
+    // cp: 'fifo' -> 'outfile': the source path is neither a regular file nor a symlink to a regular file
+    //
+    // the underling `std::fs:cp` doesn't support copying fifo on freeBSD
```
Comment (tertsdiepraam) [inline tests/by-util/test_cp.rs:2321]:
```suggestion
    // the underlying `std::fs:cp` doesn't support copying fifo on freeBSD
```

**Review by tertsdiepraam — COMMENTED** [review]
Looks good to me! Just one typo to fix

File: tests/by-util/test_cp.rs (hunk 1 above)
Comment (tertsdiepraam) [reply tests/by-util/test_cp.rs:2321]: It should also be `std::fs::copy`, right?

Comment (tertsdiepraam) [reply tests/by-util/test_cp.rs:2321]: Looks like the linux and macos use `std::io::copy`. I think this is also related to an issue Sylvestre made back in 2020: https://github.com/rust-lang/rust/issues/79390

**Comment by sylvestre** [comment]
Thanks!

**Pull request merged**